server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1275,19 +1275,20 @@ private HypervisorType getHypervisorType(VMInstanceVO vm, StoragePool srcVolumeP
         final Type hostType = srcHost.getType();
         Pair<List<HostVO>, Integer> allHostsPair = null;
         List<HostVO> allHosts = null;
+        List<HostVO> hostsWithStorageMigration = null;
```

Comment:
@davidjumani do you mean hosts for migration with storage? why another list is required here?

Reply:
Previously the `allHosts` list was altered (if storage migration was required) and then used to find the suitable hosts. Instead, now it's being cloned into `hostsWithStorageMigration` and using the cloned list to find the suitable hosts so that the original list is not altered.

Reply:
Ok @davidjumani 

```suggestion
        List<HostVO> hostsForMigrationWithStorage = null;
```